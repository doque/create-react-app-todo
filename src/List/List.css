.Todo {
  width: 400px;
  margin: 3em auto;
  border: 1px solid lightgray;
  padding: 15px;
  font: 18pt sans-serif;
}

.Todo input {
  padding: 3px;
}
