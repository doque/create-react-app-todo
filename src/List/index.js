import React, { Component } from "react";
import "./List.css";

class List extends Component {
  state = {
    todos: []
  };

  handleChange = event => {
    const newTodo = event.target.value;

    // return key was pressed
    if (newTodo !== "" && event.keyCode === 13) {
      this.setState({
        todos: this.state.todos.concat(newTodo)
      });
    }
  };

  handleSubmit(event) {
    event.preventDefault();
  }

  removeTodo = todo => {
    const { todos } = this.state;
    todos.splice(todos.indexOf(todo), 1);
    this.setState({ todos });
  };

  render() {
    return (
      <form className="Todo" onSubmit={this.handleSubmit}>

        <input
          placeholder="Neues Todo"
          type="text"
          onKeyUp={this.handleChange}
        />

        <h4>Meine Todos:</h4>

        <ul className="todos">
          {this.state.todos.length === 0 && <span>Nothing yet!</span>}

          {this.state.todos.map(todo =>
            <li className={todo.completed ? "completed" : ""}>
              {todo.text}
              <span onClick={() => this.removeTodo(todo)}> x </span>
            </li>
          )}
        </ul>
      </form>
    );
  }
}

export default List;
