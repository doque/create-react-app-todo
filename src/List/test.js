import React from 'react';
import renderer from 'react-test-renderer';

import List from './';

it('renders correctly without TODOs', () => {
    const tree = renderer.create(
        <List />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
