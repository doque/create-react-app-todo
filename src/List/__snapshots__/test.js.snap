// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without TODOs 1`] = `
<form
  className="Todo"
  onSubmit={[Function]}
>
  <input
    onKeyUp={[Function]}
    placeholder="Neues Todo"
    type="text"
  />
  <h4>
    Meine Todos:
  </h4>
  <ul
    className="todos"
  >
    <span>
      Nothing yet!
    </span>
  </ul>
</form>
`;
